File: Solutions/elixir/sieve/lib/sieve.ex
defmodule Sieve do
  @doc """
  Generates a list of primes up to a given limit.
  """
  @spec primes_to(non_neg_integer) :: [non_neg_integer]
  def primes_to(limit) do
    if limit < 2,
      do: [],
      else:
        2..limit
        |> Enum.to_list()
        |> sieve
  end

  defp sieve([]), do: []

  defp sieve([p | rest]) do
    [p | sieve(Enum.filter(rest, fn x -> rem(x, p) != 0 end))]
  end
end

File: Solutions/python/sieve/sieve.py
def primes(limit):
    # Implement the sieve of Eratosthenes here
    results = [True for _ in range(limit + 1)]
    results[0] = results[1] = False

    for i in range(2, int(limit**0.5) + 1):
        if results[i]:
            for j in range(i * i, limit + 1, i):
                results[j] = False
    return [i for i, prime in enumerate(results) if prime]
